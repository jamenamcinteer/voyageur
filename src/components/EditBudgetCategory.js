import React from "react";
import TripHeader from "./Navigation/TripHeader";
import BudgetCategoryForm from "./Forms/BudgetCategoryForm";
import { connect } from "react-redux";

const EditBudgetCategory = props => {
  const trip = props.trips.find(trip => trip._id === props.match.params.id);
  const budgetCategory = props.budgetCategories.find(
    i => i._id === props.match.params.budgetCategoryId
  );

  return (
    <div>
      {trip && (
        <div>
          <TripHeader
            title="Add Budget Category"
            theme={props.theme}
            backTo={`/trip/${props.match.params.id}`}
            trip={trip}
            budgetItems={props.budgetItems}
            expenses={props.expenses}
          />
          <BudgetCategoryForm
            theme={props.theme}
            trip={trip}
            budgetCategory={budgetCategory}
            history={props.history}
          />
        </div>
      )}
    </div>
  );
};

const mapStateToProps = (state, props) => {
  return {
    trips: state.trips,
    budgetCategories: state.budgetCategories,
    budgetItems: state.budgetItems,
    expenses: state.expenses
  };
};

export default connect(mapStateToProps)(EditBudgetCategory);
