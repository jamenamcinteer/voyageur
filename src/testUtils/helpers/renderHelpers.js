import React from "react";
import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import theme from "../../theme";
import "jest-styled-components";
import { createStore } from "redux";
import { Provider } from "react-redux";
import tripsReducer, { tripsReducerDefaultState } from "../../reducers/trips";
import { createMemoryHistory } from "history";
import { Router } from "react-router-dom";

export function renderWithTheme(component) {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>);
}

export function renderWithReduxRouterAndTheme(
  ui,
  {
    initialState,
    store = createStore(tripsReducer, tripsReducerDefaultState),
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) {
  return {
    ...render(
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <Router history={history}>{ui}</Router>
        </ThemeProvider>
      </Provider>
    ),
    store,
    history
  };
}
