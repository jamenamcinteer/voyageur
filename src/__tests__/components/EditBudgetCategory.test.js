import React from "react";
import { cleanup } from "@testing-library/react";
import { createStore } from "redux";
import "react-dates/initialize";
import EditBudgetCategory from "../../components/EditBudgetCategory";
import {
  trips,
  budgetCategories,
  budgetItems,
  expenses,
  auth
} from "../../testUtils/fixtures/mockStoreData";
import { renderWithReduxRouterAndTheme } from "../../testUtils/helpers/renderHelpers";

jest.mock("uniqid", () => {
  let value = 1;
  return () => value++;
});

afterEach(cleanup);

const store = createStore(() => ({
  trips,
  budgetCategories,
  budgetItems,
  expenses,
  auth
}));

test("can render with redux with custom store", () => {
  const { asFragment } = renderWithReduxRouterAndTheme(
    <EditBudgetCategory
      match={{ params: { id: "1", budgetCategoryId: "1" } }}
    />,
    {
      store
    }
  );
  expect(asFragment()).toMatchSnapshot();
});
