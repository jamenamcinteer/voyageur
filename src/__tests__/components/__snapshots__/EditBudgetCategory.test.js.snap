// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can render with redux with custom store 1`] = `
<DocumentFragment>
  <div>
    <div>
      <div>
        <div>
          <div
            class="sc-bdVaJa hZqfbl"
          >
            <a
              href="/trip/1"
            >
              <svg
                aria-label="Go Back"
                style="fill: #4b5c6b; width: 20px; height: 20px; float: left; margin-top: 2px;"
                version="1.1"
                viewBox="0 0 199.404 199.404"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <polygon
                    points="135.412,0 35.709,99.702 135.412,199.404 163.695,171.119 92.277,99.702 163.695,28.285 	"
                  />
                </g>
              </svg>
            </a>
            <h1
              class="sc-bwzfXH OVfcq"
            >
              Add Budget Category
            </h1>
            <button
              class="sc-htpNat fqCEYg"
            >
              <img
                alt="Account Options"
                class="sc-bxivhb jwtarI"
                src="https://picsum.photos/id/947/45/45"
              />
            </button>
          </div>
          <div
            style="position: relative;"
          >
            <div
              style="background-image: url(https://images.unsplash.com/photo-1549317487-92cec5e76645?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjc5ODc0fQ&w=300&h=100&fit=crop&crop=focalpoint); background-repeat: no-repeat; background-size: cover; filter: brightness(0.6); height: 150px;"
            />
            <div
              style="height: 150px; padding: 0px 20px; color: rgb(255, 255, 255); font-size: 1em; font-weight: normal; position: absolute; width: 100%; top: 0px;"
            >
              <h2
                class="sc-bZQynM krdFRo"
              >
                Hawaii
              </h2>
              <div
                style="display: flex; justify-content: space-between; align-items: center; padding-bottom: 10px;"
              >
                <h3
                  class="sc-gzVnrw HgWFV"
                >
                  Apr 26, 1970 - Oct 1, 8307
                </h3>
                <h3
                  class="sc-htoDjs btYDzX"
                >
                  $595 / $2000
                </h3>
              </div>
              <div
                style="width: 100%; height: 10px; position: relative; background: rgb(220, 224, 227); border-radius: 5px;"
              >
                <span
                  style="display: block; height: 100%; border-radius: 5px; position: relative; overflow: hidden; width: 29.731%; background-color: rgb(12, 124, 136);"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-kGXeez eDEywD"
      >
        <span
          class="sc-gzVnrw daRrCx"
          style="width: 29.731%; background-color: rgb(26, 174, 159);"
        />
        <div
          class="sc-dnqmqq jnbCEU"
        >
          <label
            class="sc-iwsKbI dtSDPi"
            for="1"
          >
            Name of Budget Category
          </label>
          <input
            class="sc-gZMcBi kJGmWv"
            id="1"
            placeholder="Example: Airfare, Food, Activities, etc."
            type="text"
            value="Airfare"
          />
        </div>
        <div />
        <div
          class="sc-gqjmRU jgsBPr"
        >
          <label
            class="sc-VigVT hqhiep"
            for="2"
          >
            Notes (optional)
          </label>
          <textarea
            class="sc-jTzLTM cHGIjj"
            id="2"
            type="text"
          />
        </div>
        <div
          class="sc-ckVGcZ iPSTEH"
        >
          <div>
            <button
              class="sc-jzJRlG jzvwJb"
              style="padding: 10px 0px;"
            >
              <span
                class="sc-cSHVUG gAvioZ"
              >
                Delete
              </span>
            </button>
          </div>
          <div
            class="sc-dxgOiQ lmtbyg"
          >
            <a
              backgroundcolor="transparent"
              border="0"
              buttontype="link"
              buttonwidth="auto"
              class="sc-kAzzGY fsudaQ"
              href="/trip/1"
              style="padding: 10px 0px;"
            >
              <span
                class="sc-chPdSV IudMt"
              >
                Cancel
              </span>
            </a>
            <button
              class="sc-jzJRlG hjpZFG"
            >
              <span
                class="sc-cSHVUG hQiayu"
                color="#fff"
              >
                Save
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
